scene/realistic-light-and-shadows/src/main/java/com/esri/samples/realistic_lighting_and_shadows/RealisticLightingAndShadowsController.java

```diff
@@ -0,0 +1,203 @@
+    // when the slider changes, update the hour of the day based on the value of the slider
+    timeSlider.valueProperty().addListener((observable, oldValue, newValue) -> {
+
+        // get value from the slider
+        Double timeFromSlider = timeSlider.getValue();
+        // to get minutes from timer value, split double to the two values after decimal place
+        String timeAsString = timeFromSlider.toString();
+
+        // get the hour value from the slider
+        int hourFromSlider = timeFromSlider.intValue();
+
+        if (timeAsString.length() > 4) {
+          String subString = timeAsString.substring(3, 5);
+
+          int minutes = Integer.valueOf(subString);
+          // convert figures into minutes
+          float actualMinutes = ((float) minutes / (float) 100) * (float) 60;
+          // round into an integer
+          int minuteFromSlider = Math.round(actualMinutes);
+
+          // set the calendar for given hour and minute from slider value
+          calendar.set(2018, 7, 10, hourFromSlider, minuteFromSlider);
+
+        } else {
+          calendar.set(2018, 7, 10, hourFromSlider, 00);
+        }
+        // tidy string to just return date and time (hours and minutes)
+        SimpleDateFormat dateFormat = new SimpleDateFormat("EEE MMM dd HH:mm");
+        dateFormat.setTimeZone(TimeZone.getTimeZone("PST"));
+        String dynamicDateAndTimeTidied = dateFormat.format(calendar.getTime());
+
+        // update label to reflect current date and time
+        time.setText(dynamicDateAndTimeTidied);
+
+        // set the sun time to calendar
+        sceneView.setSunTime(calendar);
+      }
+    );
+  }
+
+  /**
+   * Set labels to display on the slider.
+   */
+  private void setSliderLabels() {
+
+    timeSlider.setLabelFormatter(new StringConverter<Double>() {
+
+      @Override
+      public String toString(Double object) {
+
+        if (object == 4) return "4am";
+        if (object == 8) return "8am";
+        if (object == 12) return "Midday";
+        if (object == 16) return "4pm";
+        if (object == 20) return "8pm";
+
+        return "Midnight";
+      }
+
+      @Override
+      public Double fromString(String string) {
+        return null;
+      }
+    });
+  }
+
+  /**
+   * Set the time to display in a label.
+   */
+  public void setTimeLabel() {
+
+    // set a new calendar and add a date and time
+    calendar = new GregorianCalendar(2018, 7, 10, 12, 00, 0);
+    calendar.setTimeZone(TimeZone.getTimeZone("PST"));
+    sceneView.setSunTime(calendar);
+
+    // tidy string to just return date and time (hours and minutes)
+    SimpleDateFormat dateFormat = new SimpleDateFormat("EEE MMM dd HH:mm");
+    dateFormat.setTimeZone(TimeZone.getTimeZone("PST"));
+    String dateAndTimeTidied = dateFormat.format(calendar.getTime());
+
+    // set a label to display the tidied date and time
+    time.setText(dateAndTimeTidied);
+  }
+
+  /**
+   * Disposes application resources.
```

Comment:
I think most newer samples try and use plain english here, so let's add `Disposes of...`
```suggestion
   * Disposes of application resources.
```